NetworkExplorationCode/ExportFeatureVectors.py

```diff
@@ -0,0 +1,61 @@
+import NodeFeatureExtractor as fe
+import pandas as pd
+from tqdm import tqdm
+import graph_tool as gt
+from graph_tool import Graph
+
+
+def get_all_subgraph_features(_g: Graph, edge_features: list[str], vertex_features: list[str], sg_type: str, _d: int) \
+        -> pd.DataFrame:
+    features_df = pd.DataFrame()
+    num_v = _g.num_vertices()
+    print("proccessing graph features: ")
+    for _v in tqdm(_g.get_vertices()):
+        if fe.vertex_disconnected(_g, _v):
+            continue
+        if sg_type == "depth":
+            eg = fe.ego_net_depth_N(_g, _v, _d)  # get subgraph on depth n from vertex v
+        elif sg_type == "distance":
+            eg = fe.ego_net_distance_N(_g, _v, _d)
+        dict_features = {}
+        dict_features = fe.analyze_properties(eg, sg_type + '_4', 'e', edge_features)
+        dict_features["n_vertices"] = eg.num_vertices()
+        dict_features["n_edges"] = eg.num_edges()
+
+        dict_2_row = pd.DataFrame([dict_features])
+        features_df = pd.concat([features_df, dict_2_row], ignore_index=True)
+    return features_df
+
+
+# basic args
```

Comment:
This should be in a "if name=main" scope. If not, the function you wrote is not usable from other scripts!